Terminal.Gui/Core/View.cs

```diff
@@ -568,17 +568,19 @@ void SetTabIndex ()
 				}
 
 				width = value;
-				if (autoSize && value.Anchor (0) != TextFormatter.Size.Width
-					- (TextFormatter.IsHorizontalDirection (TextDirection)
-					&& TextFormatter.Text.Contains (HotKeySpecifier) ? 1 : 0)) {
+
+				var isValidNewAutSize = autoSize ? IsValidAutoSizeWidth (width) : false;
+
+				if (autoSize && !isValidNewAutSize) {
+					TextFormatter.AutoSize = false;
```

Comment:
Same comment/question I have for #1797: Should this throw an exception instead of changing the property?

Reply:
Sure I'll do the changes. Thanks.